.github/workflows/resolve-pr-release-versions.yml

```diff
@@ -0,0 +1,262 @@
+              data = JSON.parse(await fs.readFile(filename));
+            } catch (err) {
+              if (err.code !== 'ENOENT') {
+                throw err;
+              } else {
+                core.debug('Previous data not found, bootstrapping');
+                const resp = await fetch(BOOTSTRAP_DATA_URL);
+                data = await resp.json();
+              }
+            }
+
+            const { versions } = await ElectronVersions.create(undefined, { ignoreCache: true });
+
+            try {
+              while (true) {
+                const { rateLimit: rateLimitA, repository: { releases } } = await github.graphql(NEW_RELEASES_QUERY, { endCursor: data.endCursor, count: maxReleaseCount === 0 ? RELEASE_MAX_PAGINATION_COUNT : maxReleaseCount });
+                core.debug(rateLimitA);
+
+                if (releases.nodes.length === 0) {
+                  core.notice('No new releases to process');
+                  break;
+                }
+
+                for (const { name: tagName } of releases.nodes) {
+                  const parsedVersion = semver.parse(tagName);
+
+                  if (parsedVersion === null) {
+                    core.error(`Could not parse version from ${tagName} - skipping`);
+                    continue;
+                  } else if (parsedVersion.major < MIN_MAJOR) {
+                    core.debug(`Skipping release ${tagName} as it's before major ${MIN_MAJOR}`);
+                    continue;
+                  }
+
+                  let idx = versions.findIndex(({ version }) => `v${version}` === tagName);
+                  if (idx === -1) {
+                    core.warning(`Could not find release ${tagName} - skipping`);
+                    continue;
+                  } else if (idx === 0) {
+                    core.error(`No previous release for ${tagName} - skipping`);
+                    continue;
+                  }
+                  let previousRelease = versions[--idx];
+
+                  let endCursor = undefined;
+                  while (true) {
+                    const { rateLimit: rateLimitB, repository: { release } } = await github.graphql(RELEASE_PRS_QUERY, { endCursor, releaseHeadRef: `tags/${tagName}`, previousRelease: `refs/tags/v${previousRelease}` });
+                    core.debug(rateLimitB);
+
+                    if (release === null) {
+                      // There are occasionally missing releases which made it into index.json, so
+                      // move on to the next previous release until we're back to a valid release
+                      core.warning(`${previousRelease} is a missing release - skipping`);
+                      previousRelease = versions[--idx];
+                      continue;
+                    }
+
+                    const { compare: { commits } } = release;
+
+                    for (const commit of commits.nodes) {
+                      if (commit.associatedPullRequests.pageInfo.hasNextPage) {
```

Comment:
What number does this correspond to?

Reply:
Line 90, `associatedPullRequests(first: 20)`. This check is just a safeguard against a very unlikely edge case, where the number of `associatedPullRequests` might exceed 20 and we wouldn't be able to find the merged one without paginating. I was too lazy to do nested pagination here since it's not actually a problem so long as we keep that number high enough - 20 was arbitrary but is also limited by the overall complexity of the GraphQL query (e.g. if you make `associatedPullRequests` 100 and `labels` 100, the query gets rejected for too high of a potential complexity).

From the docs explaining why there can be more than one `associatedPullRequests`, unfortunately. We only want the merged one, but in theory there can be additional ones associated for release branches so have to handle that case.

> The merged Pull Request that introduced the commit to the repository. If the commit is not present in the default branch, additionally returns open Pull Requests associated with the commit.